### PR TITLE
BUG: Fix sort_values for empty by argument

### PR DESCRIPTION
**Pull request by dsaxton**
- [x] closes https://github.com/pandas-dev/pandas/issues/40258
- [x] tests added / passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [x] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -5833,6 +5833,8 @@ def sort_values(  # type: ignore[override]
             indexer = nargsort(
                 k, kind=kind, ascending=ascending, na_position=na_position, key=key
             )
+        else:
+            return self
```
Comment (jreback) [inline pandas/core/frame.py:5837]: return self.copy()

File: pandas/tests/frame/methods/test_sort_values.py [hunk 2]
```diff
@@ -79,6 +79,10 @@ def test_sort_values(self):
         with pytest.raises(ValueError, match=msg):
             frame.sort_values(by=["A", "B"], axis=0, ascending=[True] * 5)
 
+        # https://github.com/pandas-dev/pandas/issues/40258
```
Comment (jreback) [inline pandas/tests/frame/methods/test_sort_values.py:82]: can you add a separate test. also assert that it is *not* the same frame.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/frame.py (hunk 1 above)
Comment (dsaxton) [reply pandas/core/frame.py]: Good call, updated

File: doc/source/whatsnew/v1.2.4.rst [hunk 3]
```diff
@@ -25,7 +25,7 @@ Fixed regressions
 Bug fixes
 ~~~~~~~~~
 
--
+- Fixed bug in :meth:`DataFrame.sort_values` raising an :class:`IndexError` for empty ``by`` (:issue:`40258`)
```
Comment (jreback) [inline doc/source/whatsnew/v1.2.4.rst:28]: move to 1.3, this is an expansion not a regression.

File: pandas/tests/frame/methods/test_sort_values.py (hunk 2 above)
Comment (jreback) [reply pandas/tests/frame/methods/test_sort_values.py:82]: can you make a completely separate test

Comment (jreback) [reply pandas/tests/frame/methods/test_sort_values.py:82]: we dont' normally want to mix error checking with other things

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @dsaxton

**Pull request merged**